src/errorpage.cc

```diff
@@ -203,9 +213,18 @@ static int error_page_count = 0;
 /// \ingroup ErrorPageInternal
 static MemBuf error_stylesheet;
 
-static const char *errorFindHardText(err_type type);
 static IOCB errorSendComplete;
 
+template <class Messages>
+static const char *
+FindHardCodedTemplate(const err_type type, const Messages &messages)
+{
+    const auto foundDefault = std::find_if(messages.cbegin(), messages.cend(), [&](const HardCodedMessage &m) {
+        return m.type == type;
+    });
+    return foundDefault == messages.cend() ? nullptr : foundDefault->text;
```

Comment:
This is better implemented as a ranged-for loop. For both readability and performance.
```suggestion
    for (const auto &m : messages) {
        if (m.type == type)
            return m.text;
    }
    return nullptr;
```